### ISSUE TITLE
Can't use ipvs proxier, trying iptables proxier

### ISSUE THREAD

**Issue by roffe** [kubernetes/kubernetes#55864]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig network

**What happened**:
```
I1116 09:02:43.479194       1 feature_gate.go:156] feature gates: map[SupportIPVSProxyMode:true]
I1116 09:02:43.480561       1 feature_gate.go:156] feature gates: map[SupportIPVSProxyMode:true]
I1116 09:02:43.485844       1 iptables.go:564] couldn't get iptables-restore version; assuming it doesn't support --wait
I1116 09:02:43.492355       1 loader.go:357] Config loaded from file /etc/kubernetes/proxy-kubeconfig.yaml
W1116 09:02:43.496621       1 server_others.go:264] Can't use ipvs proxier, trying iptables proxier
I1116 09:02:43.498852       1 server_others.go:122] Using iptables Proxier.
```

**What you expected to happen**:
Kube-proxy to use IPVS experimental support

**How to reproduce it (as minimally and precisely as possible)**:
start kube-proxy with 
```
    - --feature-gates=SupportIPVSProxyMode=true
    - --proxy-mode=ipvs
```

**Anything else we need to know?**:
Host /lib/modules is mounted into the pod into the same folder as RO.

Tried preloading all the kernel modules before starting kube-proxy based on this list https://github.com/kubernetes/kubernetes/blob/release-1.8/pkg/proxy/ipvs/proxier.go#L75

Tried to exec the command at https://github.com/kubernetes/kubernetes/blob/release-1.8/pkg/proxy/ipvs/proxier.go#L622 to and verified command runs fine and all listed moduls is loaded.

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-08T18:39:33Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-08T18:27:48Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration: virtual machines in proxmox
- OS (e.g. from /etc/os-release): Container Linux by CoreOS stable (1520.8.0)
- Kernel (e.g. `uname -a`): Linux kubemaster01 4.13.9-coreos #1 SMP Thu Oct 26 03:21:00 UTC 2017 x86_64 Common KVM processor GenuineIntel GNU/Linux
- Install tools: manual kubernetes deploy
- Others:

**Comment by roffe** [comment]
The error output is with -v=10 it dose not print anything more relevant to IPVS afterwards but just proceeds to start in iptables mode

**Comment by roffe** [comment]
Suggestion:  add some more levels of output to CanUseIPVSProxier & tryIPVSProxy to be more "talky" about what it does when verbosity is turned up

**Comment by dims** [comment]
@roffe i am guessing a problem looking at `/proc/modules`, can you please try that by hand?

https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/ipvs/proxier.go#L658

**Comment by roffe** [comment]
```
core@kubemaster01 /etc/kubernetes $ docker ps
CONTAINER ID        IMAGE                                                                                                                    COMMAND                  CREATED             STATUS              PORTS                  NAMES
aa94728d6d30        gcr.io/google_containers/hyperkube-amd64@sha256:2fbd3d9ace56ac3ec2775f64f5ec5e6551ab30c5e25cc0c2633f33b15b8ca542         "/hyperkube proxy --m"   2 hours ago         Up 2 hours                                 k8s_kube-proxy_kube-
.......
core@kubemaster01 /etc/kubernetes $ docker exec -ti aa94728d6d30 cat /proc/modules | cut -f1 -d " "
ip_vs_sh
ip_vs_wrr
ip_vs_rr
ip_vs
xt_esp
drbg
seqiv
esp4
xfrm4_mode_transport
ctr
ghash_generic
gf128mul
gcm
xt_policy
iptable_mangle
xt_NFLOG
dummy
vport_vxlan
vxlan
ip6_udp_tunnel
udp_tunnel
openvswitch
nf_conntrack_ipv6
nf_nat_ipv6
nf_defrag_ipv6
nfnetlink_log
binfmt_misc
ip_set_hash_ip
xt_set
ip_set
nf_conntrack_netlink
nfnetlink
xt_statistic
xt_recent
ipt_REJECT
nf_reject_ipv4
xt_comment
xt_mark
veth
xt_nat
ipt_MASQUERADE
nf_nat_masquerade_ipv4
xfrm_user
xfrm_algo
iptable_nat
nf_conntrack_ipv4
nf_defrag_ipv4
nf_nat_ipv4
xt_addrtype
iptable_filter
xt_conntrack
nf_nat
nf_conntrack
br_netfilter
bridge
stp
llc
overlay
nls_ascii
nls_cp437
vfat
fat
virtio_balloon
i2c_piix4
evdev
psmouse
mousedev
i2c_core
button
sch_fq_codel
dm_thin_pool
dm_persistent_data
dm_bio_prison
libcrc32c
crc32c_generic
ext4
crc16
mbcache
jbd2
fscrypto
hid_generic
usbhid
hid
dm_verity
dm_bufio
sr_mod
sd_mod
cdrom
virtio_net
virtio_scsi
uhci_hcd
ehci_pci
ehci_hcd
ata_piix
usbcore
libata
scsi_mod
virtio_pci
virtio_ring
virtio
usb_common
qemu_fw_cfg
dm_mirror
dm_region_hash
dm_log
dm_mod
dax
```

**Comment by roffe** [comment]
Ok this is getting weirder. when i set 

mode: ipvs

in kubeconfig instead

then i get a new error on start:

```
core@agi-cph-kubemaster01 ~ $ docker logs 342eede5b2cb
I1116 12:50:33.983336       1 feature_gate.go:156] feature gates: map[SupportIPVSProxyMode:true]
I1116 12:50:34.001621       1 server_others.go:164] Using ipvs Proxier.
W1116 12:50:34.020545       1 proxier.go:239] IPVS scheduler not specified, use rr by default
I1116 12:50:34.020648       1 server_others.go:187] Tearing down inactive rules.
I1116 12:50:34.061939       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I1116 12:50:34.062053       1 conntrack.go:52] Setting nf_conntrack_max to 131072
I1116 12:50:34.062132       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I1116 12:50:34.062227       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
I1116 12:50:34.062455       1 config.go:202] Starting service config controller
panic: non-positive interval for NewTicker

goroutine 46 [running]:
time.NewTicker(0x0, 0x8)
	/usr/local/go/src/time/tick.go:23 +0x1b4
k8s.io/kubernetes/pkg/proxy/ipvs.(*Proxier).SyncLoop(0xc420a68f20)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/proxy/ipvs/proxier.go:736 +0x33
k8s.io/kubernetes/cmd/kube-proxy/app.(*ProxyServer).Run(0xc4208f9400, 0xc4208bb600, 0xc420208200)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:547 +0x480
k8s.io/kubernetes/cmd/kube-proxy/app.(*Options).Run(0xc420a37600, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:229 +0xa0
k8s.io/kubernetes/cmd/kube-proxy/app.NewProxyCommand.func1(0xc420314240, 0x9e02d70, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kube-proxy/app/server.go:344 +0x80
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc420314240, 0xc420913fe0, 0x0, 0x2, 0xc420314240, 0xc420913fe0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x22b
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420314240, 0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x339
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420314240, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.NewKubeProxy.func1(0xc42079d5c0, 0xc420913fe0, 0x0, 0x2, 0xc42014b020, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/kube-proxy.go:50 +0x68
main.(*HyperKube).Run.func1(0xc42079d680, 0xc42079d5c0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/hyperkube.go:184 +0x6d
created by main.(*HyperKube).Run
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/hyperkube/hyperkube.go:185 +0x588
```

**Comment by roffe** [comment]
This is the template i use to generate kube-proxy config:

```
apiVersion: componentconfig/v1alpha1
bindAddress: 0.0.0.0
clientConnection:
  acceptContentTypes: ""
  burst: 10
  contentType: application/vnd.kubernetes.protobuf
  kubeconfig: "/etc/kubernetes/proxy-kubeconfig.yaml"
  qps: 5
clusterCIDR: "${POD_NETWORK}"
configSyncPeriod: 15m0s
conntrack:
  max: 0
  maxPerCore: 32768
  min: 131072
  tcpCloseWaitTimeout: 1h0m0s
  tcpEstablishedTimeout: 24h0m0s
enableProfiling: false
featureGates: SupportIPVSProxyMode=true
healthzBindAddress: 0.0.0.0:10256
hostnameOverride: "${ADVERTISE_IP}"
iptables:
  masqueradeAll: false
  masqueradeBit: 14
  minSyncPeriod: 0s
  syncPeriod: 30s
kind: KubeProxyConfiguration
metricsBindAddress: 127.0.0.1:10249
mode: ipvs
oomScoreAdj: -999
portRange: ""
resourceContainer: /kube-proxy
udpTimeoutMilliseconds: 250ms
```

**Comment by roffe** [comment]
I think i found the problem.

I'm using a old kube-proxy config template.


after running --write-config-to on a new version i saw some new fields

```
apiVersion: componentconfig/v1alpha1
bindAddress: 0.0.0.0
clientConnection:
  acceptContentTypes: ""
  burst: 10
  contentType: application/vnd.kubernetes.protobuf
  kubeconfig: ""
  qps: 5
clusterCIDR: ""
configSyncPeriod: 15m0s
conntrack:
  max: 0
  maxPerCore: 32768
  min: 131072
  tcpCloseWaitTimeout: 1h0m0s
  tcpEstablishedTimeout: 24h0m0s
enableProfiling: false
featureGates: ""
healthzBindAddress: 0.0.0.0:10256
hostnameOverride: ""
iptables:
  masqueradeAll: false
  masqueradeBit: 14
  minSyncPeriod: 0s
  syncPeriod: 30s
ipvs:
  minSyncPeriod: 0s
  scheduler: ""
  syncPeriod: 0s
kind: KubeProxyConfiguration
metricsBindAddress: 127.0.0.1:10249
mode: ""
oomScoreAdj: -999
portRange: ""
resourceContainer: /kube-proxy
udpTimeoutMilliseconds: 250ms
```

results inc

**Comment by roffe** [comment]
Success!

I1116 13:04:38.068488       1 feature_gate.go:156] feature gates: map[SupportIPVSProxyMode:true]
I1116 13:04:38.085364       1 server_others.go:164] Using ipvs Proxier.
W1116 13:04:38.103724       1 proxier.go:239] IPVS scheduler not specified, use rr by default
I1116 13:04:38.103784       1 server_others.go:187] Tearing down inactive rules.
I1116 13:04:38.155993       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I1116 13:04:38.156078       1 conntrack.go:52] Setting nf_conntrack_max to 131072
I1116 13:04:38.156129       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I1116 13:04:38.156181       1 conntrack.go:98] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
I1116 13:04:38.156496       1 config.go:202] Starting service config controller
I1116 13:04:38.156559       1 controller_utils.go:1041] Waiting for caches to sync for service config controller
I1116 13:04:38.156749       1 config.go:102] Starting endpoints config controller
I1116 13:04:38.156766       1 controller_utils.go:1041] Waiting for caches to sync for endpoints config controller
I1116 13:04:38.256937       1 controller_utils.go:1048] Caches are synced for endpoints config controller
I1116 13:04:38.256947       1 controller_utils.go:1048] Caches are synced for service config controller

However the default syncPeriod: 0s will give the panic `panic: non-positive interval for NewTicker`

**Comment by roffe** [comment]
Closing as this issue is resolved and the cause was configuration error on my side. However opened a new issue https://github.com/kubernetes/kubernetes/issues/55886 as I discovered a config bug during the work with this

**Comment by ivanilves** [comment]
Thanks for opening this issue @roffe 

You saved me **lots** of time by posting your flow :clap:
